docs: remove Component suffix from examples

SkyZeroZx opened the pull request:
This PR is part of the work to remove the `Component` suffix from documentation examples, as discussed in #66221.

As an initial step, it updates to align examples with the current Angular Style Guide. Further guides will be updated incrementally.

Guides updates:
- Drag and Drop 
- DI

JeanMeche commented:
From my understanding, the style guide doesn't only recommend to drop the component suffix but suggest to give more meaningful names to component classes. 

Cases where you renamed `MyButtonComponent` component to `MyButton` or `HeroListPageComponent` to `HeroListPage` make perfect sense but other changes are less convincing.
Renaming `ProfileComponent` to `Profile` is exactly why the community didn't like the new recommendation in the first place. Classes should bear a meaningful name. And a single word like`Profile` is simply too vague. 

And as such I would consider the PR way to large to be thouroughly reviewed, since we cannot accept a change that only removes the component suffix.

SkyZeroZx commented:
> From my understanding, the style guide doesn't only recommend to drop the component suffix but suggest to give more meaningful names to component classes.
> 
> Cases where you renamed `MyButtonComponent` component to `MyButton` or `HeroListPageComponent` to `HeroListPage` make perfect sense but other changes are less convincing. Renaming `ProfileComponent` to `Profile` is exactly why the community didn't like the new recommendation in the first place. Classes should bear a meaningful name. And a single word like`Profile` is simply too vague.
> 
> And as such I would consider the PR way to large to be thouroughly reviewed, since we cannot accept a change that only removes the component suffix.


I’ll split the changes into smaller follow-up PRs.

This PR limits component suffix removals to file names, the DI guides, and the Drag and Drop guide.

It also fixes a few syntax highlighting issues found along the way.

Improving consistency helps downstream tooling and AI generate files using the expected conventions.

Decorators and headers are updated as well, for example renaming `app.component.ts` to `app.ts`

josephperrott reviewed (approved):
LGTM

JeanMeche commented:
Can you please address the conflict. Thank you.

SkyZeroZx committed:
docs: remove Component suffix from examples in DI & Drag and Drop

Updates documentation examples to remove the `Component` suffix, aligning them with the current Angular Style Guide.
